src/main/java/com/mraof/minestuck/blockentity/IntellibeamLaserstationBlockEntity.java

```diff
@@ -0,0 +1,196 @@
+package com.mraof.minestuck.blockentity;
+
+import com.mraof.minestuck.alchemy.AlchemyHelper;
+import com.mraof.minestuck.block.MSBlocks;
+import com.mraof.minestuck.block.machine.IntellibeamLaserstationBlock;
+import com.mraof.minestuck.item.MSItems;
+import com.mraof.minestuck.util.MSSoundEvents;
+import net.minecraft.ChatFormatting;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.protocol.Packet;
+import net.minecraft.network.protocol.game.ClientGamePacketListener;
+import net.minecraft.network.protocol.game.ClientboundBlockEntityDataPacket;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.Containers;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+
+public class IntellibeamLaserstationBlockEntity extends BlockEntity
+{
+	protected ItemStack card = ItemStack.EMPTY;
+	protected int EXP_LEVEL_CAPACITY = 10;
+	protected int EXPERIENCE_LEVEL = 0;
+	protected float SOUND_SCALER = 0F;
+	protected static int WAIT_TIMER = 0;
+	
+	public IntellibeamLaserstationBlockEntity(BlockPos pos, BlockState state)
+	{
+		super(MSBlockEntityTypes.INTELLIBEAM_LASERSTATION.get(), pos, state);
+	}
+	
+	public void onRightClick(Player player)
+	{
+		ItemStack heldItem = player.getMainHandItem();
+		
+		if(WAIT_TIMER > 0)
+		{
+			return;
+		}
+		if(!card.isEmpty() && player.isShiftKeyDown())
+		{
+			takeCard(player);
+			WAIT_TIMER = 10;
+			return;
+		}
+		if(AlchemyHelper.isReadableCard(card))
+		{
+			this.level.playSound(null, this.worldPosition, MSSoundEvents.INTELLIBEAM_LAZERSTATION_REMOVE_CARD.get(), SoundSource.BLOCKS, 0.5F, 0.1F);
+			WAIT_TIMER = 10;
+			return;
+		}
+		if(card.isEmpty() && !AlchemyHelper.isReadableCard(card))
+		{
+			tryInsertCard(heldItem);
+			WAIT_TIMER = 10;
+			return;
+		}
```

Comment:
a dedicated GUI, isn't needed since it's so conceptually simple. but if clarity is a worry then a small HWYLA styled gui that appears next to your cursor for each of the alchemy related items could do the trick. simply having an "experience gauge" would be all it takes. although i'm of the opinion that by the time the player will be able to get the intellibeam laserstation, Experience won't be much of a concern. So letting players stumble into an answer through experimentation here seems fine. but the aforementioned GUI would be a nice aesthetic touch

Reply:
besides that, empty captcha cards do have codes, they're just 0000000.

Reply:
the third return should prevent readable cards from actually being able to have experience added